src/qt/bitcoin.cpp

```diff
@@ -393,6 +394,7 @@ WId BitcoinApplication::getMainWinId() const
 static void SetupUIArgs()
 {
     gArgs.AddArg("-choosedatadir", strprintf("Choose data directory on startup (default: %u)", DEFAULT_CHOOSE_DATADIR), ArgsManager::ALLOW_ANY, OptionsCategory::GUI);
+    gArgs.AddArg("-guisettings=<path>", "Choose a custom data directory especially for the Qt Settings", ArgsManager::ALLOW_ANY, OptionsCategory::GUI);
```

Comment:
I would suggest to call it -guisettings**dir**, to be consistent with other *dir's options we have. Currently, it makes me fee as I am expected to write full path to the `.ini` file itself, though this is not the case, it's the parent(-parent) directory..

Reply:
I think path value suggestion says that it is a path to a dir. Otherwise it would be `<file>`

Reply:
"path" can mean file path or directory path.
> A path, the general form of the name of a file or directory...

https://en.wikipedia.org/wiki/Path_(computing)


This is not hard argument but simply I dont't see the reason to make this option stand out from others. We have bunch of `*dir` options, not sure why this should not have same suffix.

Reply:
@ryanofsky @laanwj what's your view about `-guisettings` vs `-guisettingsdir` (more consistent) ?

Reply:
> @ryanofsky @laanwj what's your view about `-guisettings` vs `-guisettingsdir` (more consistent) ?

If it actually is required to be a filename not a directory name then dir suffix could be better, but from the qt documentation, I got the impression that it would accept either an ini file path or a directory path. I keep asking for more documentation in every comment I make on this PR. I just think whatever the behavior is it should be documented.

Reply:
@ryanofsky: 

- if I pass `-guisettings=/tmp/mysettings`, it creates `/tmp/mysettings/Bitcoin/Bitcoin-Qt-regtest.ini` file.
- if I pass `-guisettings=/tmp/mysettings.ini` (file nor directory does not exist), it creates `/tmp/mysettings.ini/Bitcoin/Bitcoin-Qt-regtest.ini`
- if I pass `-guisettings=/tmp/mysettings.ini` after  `touch /tmp/mysettings.ini`, no directories are created, but neither `mysettings.ini` is used as settings file - it's empty. Looks like a silent failure.

So it seems that for Bitcoin it works as a parent directory, as `Bitcoin` subdirectory is always tried to be created within specified `guisettings` path.

Reply:
Thanks for testing! If that's the case, I think it'd be good to add the dir suffix, document that the path is supposed to be a directory, and also adding error checking for the silent failure case. Error checking could probably be done by calling [QSettings::sync()](https://doc.qt.io/qt-5/qsettings.html#sync) and checking [QSettings::status()](https://doc.qt.io/qt-5/qsettings.html#status) returns [QSettings::NoError](https://doc.qt.io/qt-5/qsettings.html#Status-enum).

Reply:
Ok, will switch to `-guisettingsdir`